src/components/Form/Textarea.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { Component, ChangeEvent, FocusEvent } from 'react';
+import {
+  HtmlLabel,
+  HtmlTextarea,
+  HtmlTextareaProps,
+  HtmlSpan,
+} from '../../reset';
+import { VisuallyHidden } from '../Visually-hidden/Visually-hidden';
+import { Paragraph } from '../Paragraph/Paragraph';
+import classnames from 'classnames';
+import styled from 'styled-components';
+import { idGenerator } from '../../utils/uuid';
+
+const baseClassName = 'fi-textarea';
+const textareaClassNames = {
+  label: `${baseClassName}_label`,
+  optionalText: `${baseClassName}_optionalText`,
+  textarea: `${baseClassName}_textarea`,
+  hintText: `${baseClassName}_hintText`,
+  statusText: `${baseClassName}_statusText`,
+};
+
+export interface TextareaProps extends HtmlTextareaProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable usage */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: () => void;
+  /** To execute on textarea text change */
+  onChange?: (event: ChangeEvent<HTMLTextAreaElement>) => void;
+  /** To execute on textarea text onBlur */
+  onBlur?: (event: FocusEvent<HTMLTextAreaElement>) => void;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: 'hidden' | 'visible';
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Text content for textarea */
+  children?: string;
+  /** Hint text to be shown below the component */
+  hintText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: 'default' | 'error';
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** Resize mode of the textarea
+      'both' | 'vertical' | 'horizontal' | 'none'
+      @default 'vertical' 
+   */
+  resize?: 'both' | 'vertical' | 'horizontal' | 'none';
+  optionalText?: string;
```

Comment:
Took me a moment to grasp what this is. It needs to be documented at the very least. Maybe we should/could come up with a common naming and practice for this prop to use across the components.

Reply:
Good find, it seems to be missing JSDoc.

Yes, this **OptionalText** is one of the "internal components"-idea that is going to be common in multiple components.

Reply:
JSDoc is now added.
Also changed that if the `labelMode` is **"hidden"** then the optional text is also added. Now it did not exist.